File: ArraysAndHashing/ContainsDuplicated.cs
namespace ConatinsDuplicated;

/*
    * 217. Contains duplicated
    * Given an array nums, return true if any value appears at least twice in the array,
    * and return false if every element is distinct
    */
public class ContainsDuplicated
{
    public static void Main(string[] args) {}
    
    //Space: O(n), time: O(n)
    public bool Solution(int[] nums)
    {
        /*
        var set = new HashSet<int>();
        foreach (var num in nums)
        {
            if (!set.Add(num)) return true;
        }
        return false;
        */
        //one liner
        return new HashSet<int>(nums).Count != nums.Length;
    }
}

File: ArraysAndHashing/containsDuplicated.js
/**
* 217. Contains duplicated.
* https://leetcode.com/problems/contains-duplicate/
* Given an integer array nums, return true if any value appears at least twice in the array, or false if
* every element is distinct.
* Example 1: 
* Input: [1,2,3,3]
* Output: true
* */

const input = [1,2,3,3];

function hasDuplicate(nums) {
    return new Set(nums).size != nums.length;
}

console.log('Input: ', input);
console.log('Output:', hasDuplicate(input));